hal/armv8m/nrf/91/nrf91.c

```diff
@@ -47,13 +50,37 @@ enum { fpu_cpacr = 34, fpu_fpccr = 141, fpu_fpcar, fpu_fpdscr };
 /* clang-format on */
 
 
+extern void _interrupts_nvicSystemReset(void);
+
+
 /* platformctl syscall */
 
 
-/* TODO: add platformctl implementation */
 int hal_platformctl(void *ptr)
 {
-	return -ENOSYS;
+	platformctl_t *data = ptr;
+	int ret;
+
+	switch (data->type) {
+		case pctl_reboot:
+			if (data->action == pctl_set) {
+				if (data->reboot.magic == PCTL_REBOOT_MAGIC) {
+					_interrupts_nvicSystemReset();
+				}
+			}
+			else {
+				if (data->action == pctl_get) {
+					data->reboot.reason = nrf91_common.resetFlags;
+					ret = 0;
+				}
+			}
```

Comment:
`ret` might be uninitialized in this function, no warn because nrf is not enabled on CI

Reply:
Done, thanks!

Reply:
Now you return -EINVAL on pctl_get

Reply:
What do you mean? It's set to 0 then

Reply:
Sorry, it's ok